tls/s2n_client_hello.c

```diff
@@ -416,26 +416,22 @@ int s2n_client_hello_send(struct s2n_connection *conn)
     }
 
     struct s2n_stuffer *out = &conn->handshake.io;
-    struct s2n_stuffer client_random = {0};
     uint8_t client_protocol_version[S2N_TLS_PROTOCOL_VERSION_LEN] = {0};
 
-    struct s2n_blob b = {0};
-    POSIX_GUARD(s2n_blob_init(&b, conn->handshake_params.client_random, S2N_TLS_RANDOM_DATA_LEN));
-    /* Create the client random data */
-    POSIX_GUARD(s2n_stuffer_init(&client_random, &b));
-
-    struct s2n_blob r = {0};
-    POSIX_GUARD(s2n_blob_init(&r, s2n_stuffer_raw_write(&client_random, S2N_TLS_RANDOM_DATA_LEN), S2N_TLS_RANDOM_DATA_LEN));
-    POSIX_ENSURE_REF(r.data);
-    POSIX_GUARD_RESULT(s2n_get_public_random_data(&r));
-
     uint8_t reported_protocol_version = MIN(conn->client_protocol_version, S2N_TLS12);
     client_protocol_version[0] = reported_protocol_version / 10;
     client_protocol_version[1] = reported_protocol_version % 10;
     conn->client_hello_version = reported_protocol_version;
```

Comment:
can you explain what this is doing?

Reply:
It's how you serialize / deserialize the TLS version. One byte for the ones place, one for the tens place.

So TLS1.2 is version "33", but instead of writing it as "0x00 0x21", it's written as "0x03 0x03".